packages/compiler/core/checker.ts

```diff
@@ -886,7 +896,10 @@ export function createChecker(program: Program): Checker {
           continue;
         }
 
-        const newPropType = cloneType(prop, { sourceProperty: prop, model: intersection });
+        const newPropType = cloneType(prop, {
```

Comment:
I wonder if we need to re-execute property decorators here after cloning so that any like `@path`, `@key`, etc get stored for the new property in the requisite decorator state objects?